### PR TITLE
Swap out hyperparams in strategies, mcts and set up unit test flag co…

### PR DESCRIPTION
**Pull request by brilee**
…mpatibility

Check it out: some incremental switching out of magic constants for flags.

### CODE REVIEW

File: strategies.py [hunk 1]
```diff
@@ -74,7 +78,7 @@ def __init__(self, network, seconds_per_move=5, simulations_per_move=0,
         if two_player_mode:
             self.temp_threshold = -1
         else:
-            self.temp_threshold = TEMPERATURE_CUTOFF
+            self.temp_threshold = FLAGS.softpick_move_cutoff
```
Comment (sethtroisi) [inline strategies.py]: unrelated temp_threshold appears to be used inconsistently

https://github.com/tensorflow/minigo/blob/master/strategies.py#L135
https://github.com/tensorflow/minigo/blob/master/strategies.py#L156

I think that L156 should be >=

File: tests/test_strategies.py [hunk 2]
```diff
@@ -16,11 +16,13 @@
 import unittest.mock as mock
 import numpy as np
 
+from absl import flags
+
 import coords
 import go
 from go import Position
+import mcts
```
Comment (sethtroisi) [inline tests/test_strategies.py]: unused

File: strategies.py (hunk 1 above)
Comment (amj) [reply strategies.py]: it was fixed in the v5-running head.

**Comment by amj** [comment]
ok2merge by me

File: strategies.py (hunk 1 above)
Comment (brilee) [reply strategies.py]: done

**Comment by sethtroisi** [comment]
I'm pretty sure this broke tests.

I'm getting errors in master about "absl.flags._exceptions.UnparsedFlagAccessError: Trying to access flag --softpick_move_cutoff before flags were parsed."

**Comment by brilee** [comment]
Which entrypoint are you using? All the entry points now need to have a flags.FLAGS() call somewhere in order to initialize the flag default values and potentially parse cmd line flags. I added them to the entry points I could think of, but I probably missed one.

**Comment by sethtroisi** [comment]
`BOARD_SIZE=9 python -m unittest`

**Comment by brilee** [comment]
Ah, okay. You should use ./test.sh to execute the test runner w/ flag support.

**Comment by sethtroisi** [comment]
Okay, will do from now on thanks,
We should probably update the README file

**Comment by amj** [comment]
+1 let's update the README file asap

File: strategies.py [hunk 3]
```diff
@@ -18,18 +18,22 @@
...
-from mcts import MCTSNode, MAX_DEPTH
 
+import coords
 import go
+import mcts
+import sgf_wrapper
 
-# When to do deterministic move selection.  ~30 moves on a 19x19, ~8 on 9x9
-TEMPERATURE_CUTOFF = int((go.N * go.N) / 12)
+flags.DEFINE_integer('softpick_move_cutoff', (go.N * go.N // 12) // 2 * 2,
```
Comment (brilee) [inline strategies.py]: cc @tommadams to fix this in C++ player

**Pull request merged**